code/game/machinery/constructable_frame.dm

```diff
@@ -180,7 +180,11 @@
 						break
 				if(component_check)
 					P.play_tool_sound(src)
-					var/obj/machinery/new_machine = new circuit.build_path(loc, ndir=dir) //NSV13 - added ndir argument
+					var/obj/machinery/new_machine = new circuit.build_path(loc)
+					//NSV13 - start directional machine handling
+					if(circuit.directional_build == TRUE)
+						new_machine.setDir(dir)

```

Comment:
Would it not make more sense to override the "on_construction()" method to do all this?

Reply:
Good call, fixed